lib/internal/http2/compat.js

```diff
@@ -59,6 +60,9 @@ function assertValidHeader(name, value) {
     err = new ERR_HTTP2_PSEUDOHEADER_NOT_ALLOWED();
   } else if (value === undefined || value === null) {
     err = new ERR_HTTP2_INVALID_HEADER_VALUE(value, name);
+  } else if (name === constants.HTTP2_HEADER_CONNECTION) {
+    headerMessageWarn(name);
```

Comment:
This will still set the header but to a blank value. Instead it should not set it at all which can't be done by modifying this function.

Reply:
Ignoring this header in `[kSetHeader]`